lib/pbench/server/api/resources/intake_base.py

```diff
@@ -133,36 +126,28 @@ def process_metadata(metas: list[str]) -> JSONOBJECT:
             )
         return metadata
 
-    def _backup_tarball(self, tarball_path: Path, md5_path: Path) -> Backup:
-        """Helper function which creates a backup copy of a tarball/md5 file-pair"""
-
-        # The type returned by shutil.copy() depends on the types of the inputs;
-        # since we want Path's, construct them explicitly.
+    def _backup_tarball(self, tarball_path: Path, md5_str: str) -> Path:
+        """Helper function which creates a backup copy of a tarball file"""
+        backup_target = self.backup_dir / md5_str / tarball_path.name
         try:
-            md5_backup = Path(shutil.copy(md5_path, self.backup_dir))
+            backup_target.parent.mkdir(exist_ok=True)
```

Comment:
I think it is better _not_ to raise an error if the directory already exists -- if we manage to do an upload of an existing dataset, then overwriting the backup with a fresh one seems like an evil less than failing the upload because we're worried about a conflicting backup -- do you concur?

Reply:
The worst case is a "real live hash collision" where it's supposed to be a different dataset. Overwriting the old backup would be bad. However, in theory, the intake would have failed due to the duplicate MD5 way before we get to backup and the behavior here isn't important. Whether we need to support multiple distinct resources that hash to the same MD5 is an entirely different discussion; but if we were to have that discussion and decide we need to support it, your new code is good so long as they have distinct names, and I'm not prepared to worry about it now.

Reply:
> Overwriting the old backup would be bad.

Only if the datasets are actually different.  And, if the datasets have different filenames, then we won't overwrite the previous one -- we'll have them both side-by-side in the same directory.  (If they have the same hash and the same name, I think it's very unlikely that they would be different.)

> in theory, the intake would have failed due to the duplicate MD5 way before we get to backup and the behavior here isn't important.

Agreed...which is why I'm inclined to be permissive and accommodating.  (I think this will only matter if we do something like blow away the database and re-upload files after forgetting to clear out the backup tree.)

> Whether we need to support multiple distinct resources that hash to the same MD5 is an entirely different discussion

Indeed...and I would say the odds are against us ever needing it.  😀